sympy/core/mul.py

```diff
@@ -1150,12 +1150,19 @@ def _eval_is_negative(self):
         return (-self).is_positive
 
     def _eval_is_odd(self):
+        from sympy import sqrt
         is_integer = self.is_integer
 
         if is_integer:
             r, acc = True, 1
+            _iter = -1
             for t in self.args:
+                _iter += 1
                 if not t.is_integer:
+                    if t.is_Rational:
+                        other = self.args[_iter ^ 1]
+                        if t.as_numer_denom()[1].is_even and sqrt(other).is_even:
```

Comment:
I'm not confident about this `t.as_numer_denom()[1].is_even`(case: n=2, den=4) but somehow it's working.